utils/build-script

```diff
@@ -1200,8 +1230,45 @@ details of the setups of other systems or automated environments.""")
         source_root=SWIFT_SOURCE_ROOT,
         build_root=os.path.join(SWIFT_BUILD_ROOT, args.build_subdir))
 
-    if args.clean and os.path.isdir(workspace.build_root):
-        shutil.rmtree(workspace.build_root)
+    # Unset environment variables that might affect how tools behave.
+    for v in [
+            'MAKEFLAGS',
+            'SDKROOT',
+            'MACOSX_DEPLOYMENT_TARGET',
+            'IPHONEOS_DEPLOYMENT_TARGET',
+            'TVOS_DEPLOYMENT_TARGET',
+            'WATCHOS_DEPLOYMENT_TARGET']:
+        os.environ.pop(v, None)
+
+    # Prepare Ninja build if required
+    ninja_build = None
+    if args.build_ninja:
+        ninja_source_dir = workspace.source_dir("ninja")
+        if not os.path.exists(ninja_source_dir):
+            print_with_argv0("Can't find source directory for ninja "
+                             "(tried %s)" % (ninja_source_dir))
+            return 1
+
+        ninja_build = products.Ninja(
+            args=args,
+            toolchain=toolchain,
+            source_dir=ninja_source_dir,
+            build_dir=workspace.build_dir("build", "ninja"))
+
+    if args.show_sdks:
+        swift_build_support.debug.print_xcodebuild_versions()
+
+    # Clean build direcotry if requested.
+    if args.clean:
+        shell.rmtree(workspace.build_root)
```

Comment:
The `if args.show_sdks:` and `if args.clean:` statements seem to be interrupting the flow here.  Is there an ordering dependency?  If not, could you reorder this code so that first we show SDKs, then clean, then configure ninja and build it?